sign.go

```diff
@@ -458,18 +496,41 @@ func (m *SignMessage) Verify(external []byte, verifiers ...Verifier) error {
 		return fmt.Errorf("%d verifiers for %d signatures", len(verifiers), len(m.Signatures))
 	}
 
+	payload, err := resolvePayload(m.Payload, detachedPayload)
+	if err != nil {
+		return err
+	}
+
 	// populate common parameters
 	var protected cbor.RawMessage
-	protected, err := m.Headers.MarshalProtected()
+	protected, err = m.Headers.MarshalProtected()
 	if err != nil {
 		return err
 	}
 
 	// verify message accordingly
 	for i, signature := range m.Signatures {
-		if err := signature.Verify(verifiers[i], protected, m.Payload, external); err != nil {
+		if err := signature.Verify(verifiers[i], protected, payload, external); err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+func resolvePayload(payloads ...[]byte) ([]byte, error) {
+	var payload []byte
+	for _, candidatePayload := range payloads {
+		if candidatePayload != nil {
+			if payload == nil {
+				payload = candidatePayload
+			} else {
+				return nil, ErrMultiplePayloads
+			}
```

Comment:
If you reverse the condition the code looks cleaner.
```suggestion
			if payload != nil {
				return nil, ErrMultiplePayloads
			}
			payload = candidatePayload
```